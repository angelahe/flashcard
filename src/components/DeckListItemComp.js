import React from 'react';
import PropTypes from 'prop-types';
import '../styles/flashcard.css';
import editbtn from '../img/buttons/edit_FFFFFF.png';
import deletebtn from '../img/buttons/delete_FFFFFF.png';

const DeckListItemComp = (props) => {
  const {
    deck,
    onDeckDelete,
    onDeckSelect,
    onDeckEdit,
  } = props;

  function handleItemClick() {
    console.log('in handleItemClick');
    // do something, then
    onDeckSelect(deck.deck_id);
  }

  function handleEditClick() {
    console.log('in handle edit click');
    onDeckEdit(deck.deck_id);
  }

  function handleDeleteClick() {
    console.log('in handle delete click');
    onDeckDelete(deck.deck_id);
  }

  return (
    <div
      className="DeckListItemComp ItemBox AppListItem"
      role="button"
      tabIndex="0"
    >
      <span
        className="DeckListItemText DetailText"
        role="button"
        tabIndex="0"
        onClick={handleItemClick}
        onKeyPress={handleItemClick}
      >{deck.deck_id}
      </span>
      <button className="DeckListItemEdit AppBtn" type="button" onClick={handleEditClick}>
        <img className="btnImg" src={editbtn} alt="Edit" />
      </button>
      <button className="DeckListItemDelete AppBtn Delete" type="button" onClick={handleDeleteClick}>
        <img className="btnImg" src={deletebtn} alt="Delete" />
      </button>
      <hr />
    </div>
  );
};

DeckListItemComp.defaultProps = {
  onDeckSelect: () => { },
  onDeckEdit: () => { },
  onDeckDelete: () => { },
};

DeckListItemComp.propTypes = {
  deck: PropTypes.shape({
    deck_id: PropTypes.string.isRequired,
  }).isRequired,
  onDeckSelect: PropTypes.func,
  onDeckEdit: PropTypes.func,
  onDeckDelete: PropTypes.func,
};
export default DeckListItemComp;
