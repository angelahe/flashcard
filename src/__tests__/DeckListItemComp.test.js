import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import DeckListItemComp from '../components/DeckListItemComp';

const deck = { deck_id: 'ABCD-1234' };

it('renders without crashing', () => {
  shallow(<DeckListItemComp deck={deck} />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DeckListItemComp deck={deck} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('DeckListItemComp renders correctly', () => {
  const tree = renderer.create(<DeckListItemComp deck={deck} />).toJSON();
  expect(tree).toMatchSnapshot();
});
