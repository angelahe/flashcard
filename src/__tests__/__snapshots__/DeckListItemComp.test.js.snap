// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckListItemComp renders correctly 1`] = `
<div
  className="ItemBox AppListItem"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="DetailText"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    ABCD-1234
  </span>
  <button
    className="AppBtn"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Add"
      className="btnImg"
      src="edit_FFFFFF.png"
    />
  </button>
  <button
    className="AppBtn Delete"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Delete"
      className="btnImg"
      src="delete_FFFFFF.png"
    />
  </button>
  <hr />
</div>
`;
